base/task.jl

```diff
@@ -1181,3 +1205,20 @@ if Sys.iswindows()
 else
     pause() = ccall(:pause, Cvoid, ())
 end
+
+# update the `cpu_time_ns` field of `t` to include the time since it last started running.
+function record_cpu_time!(t::Task)
+    if t.metrics_enabled && !istaskdone(t)
+        @atomic :monotonic t.cpu_time_ns += delta_time_ns() - t.last_started_running_at
```

Comment:
Do we have to share the implementation between C and Julia?

Can we just export a single global and access it from Julia?

Reply:
Good idea, that's probably faster.
(So duplicate the `delta_time_ns()` implementation, you mean?)

We will have to ccall `jl_hrtime()` either way though, so i'm not sure if it will actually save us anything?

Reply:
Yeah, but that is one function call, instead of one and then another. Since `jl_hrtime` won't be inlined.

Reply:
💡 yep, thanks, makes sense! 👍 I'll do that then.

Reply:
moot, because this change is reverted